### PR TITLE
Purchase component refactor

### PR DESCRIPTION
**Pull request by iwonakulacz**

### CODE REVIEW

File: src/containers/OfferContainer/OfferWrapper.js [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+  const [errorMsg, setErrorMsg] = useState();
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [couponDetails, setCouponDetails] = useState(null);
+
+  const createOrderHandler = () => {
+    createOrder(offerId).then(orderDetailsResponse => {
+      const { errors } = orderDetailsResponse;
+      if (orderDetailsResponse.errors.length) {
```
Comment (pawelacio) [inline src/containers/OfferContainer/OfferWrapper.js:56]: Maybe we can use that errors from the previous line `if (errors.length) {` ?

File: src/containers/OfferContainer/OfferWrapper.js [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+          couponLoading: false,
+          showMessage: true,
+          message: 'Your coupon has been applied!',
+          messageType: MESSAGE_TYPE_SUCCESS
+        });
+      }
+    });
+  };
+
+  useState(() => {
```
Comment (pawelacio) [inline src/containers/OfferContainer/OfferWrapper.js:137]: Shouldn't that be `useEffect`?

**Review by mirydz — APPROVED** [review]

File: src/containers/OfferContainer/OfferContainer.js [hunk 3]
```diff
@@ -11,239 +10,235 @@ import {
...
   onSuccess: () => {},
   urlProps: {},
   t: k => k
 };
 
-export { OfferContainer as PureOfferContainer };
-
 // export default withTranslation()(labeling()(OfferContainer));
-export default OfferContainer;
+export default OfferWrapper;
```
Comment (pawelacio) [inline src/containers/OfferContainer/OfferContainer.js:244]: If thats OfferWrapper then we also can change filename

Comment (iwonakulacz) [reply src/containers/OfferContainer/OfferContainer.js]: I prefer to keep the old name 'OfferContainer', so I rename the OfferWrapper to OfferContainer

**Review by pawelacio — APPROVED** [review]

**Pull request merged**